<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>NPLUG • claman</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="NPLUG">
<meta property="og:description" content="claman">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">claman</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/NPLUG.html">NPLUG</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="NPLUG_files/header-attrs-2.7/header-attrs.js"></script><script src="NPLUG_files/kePrint-0.0.1/kePrint.js"></script><link href="NPLUG_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>NPLUG</h1>
            
      
      
      <div class="hidden name"><code>NPLUG.Rmd</code></div>

    </div>

    
    
<p>The NPLUG (<strong>N</strong>itrogen, <strong>P</strong>hosphorous, <strong>L</strong>eucine, <strong>U</strong>racil, or <strong>G</strong>lucose) experiment of <a href="https://pubmed.ncbi.nlm.nih.gov/19889834/">Boer et al. 2010</a> explored the impacts of nutrient limitation and growth rate on the yeast’ metabolome.</p>
<p>Growth conditions in chemostat culture (indefinite exponential growth) followed a full factorial experimental design, such that the 25 primary experimental conditions contained all pairs of:</p>
<ul>
<li>5 limiting nutrients: nitrogen, phosphorous, and glucose, as well as leucine and uracil in auxotrophs.</li>
<li>5 dilution rates (growth rate / ln(2)): from 0.05-0.3 hrs<sup>-1</sup>.</li>
</ul>
<p>In total, 136 biological samples were collected. Two extraction methods were used on each culture (pellet- or filter-based) and each extraction method had two technical replicates. In addition, each batch of experimental cultures was paired with a phosphate-limited slow growth (DR = 0.05 hrs<sup>-1</sup>) reference culture to correct for month-to-month variability.</p>
<p>This study demonstrates the nature of nutrient limitation massively impacts the metabolome, with some growth-limiting metabolites becoming greatly depleted and other metabolites appearing to overflow. This vignette reproduces some of the major findings of this study using the standard functionality built into <strong>claman</strong>.</p>
<hr>
<div id="data-import" class="section level1">
<h1 class="hasAnchor">
<a href="#data-import" class="anchor"></a>Data Import</h1>
<p>claman is meant to nicely interface with .mzrollDB SQLite databases produced by <a href="https://github.com/eugenemel/maven">MAVEN</a>, the Metabolomics Analysis and Visualization ENgine. Accordingly, we can start with a .mzrollDB file containing a stripped-down version of the NPLUG data, which bas been bundled with the claman package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">claman</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>

<span class="va">mzroll_db_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nplug_mzroll.html">nplug_mzroll</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>This database contains tables of peakgroups (ions with characteristic m/z and rt), samples, and peaks (measurements of peakgroups in each sample). It does not include meta-data about metabolites (such as metabolites’ pathways), nor samples (such as the experimental design), so this information will be added later.</p>
<div id="read--mzrolldb" class="section level2">
<h2 class="hasAnchor">
<a href="#read--mzrolldb" class="anchor"></a>Read .mzrollDB</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mzroll_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/process_mzroll.html">process_mzroll</a></span><span class="op">(</span><span class="va">mzroll_db_path</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<caption>
Features table
</caption>
<thead><tr>
<th style="text-align:left;">
groupId
</th>
<th style="text-align:left;">
compoundName
</th>
<th style="text-align:left;">
smiles
</th>
<th style="text-align:left;">
adductName
</th>
<th style="text-align:left;">
tagString
</th>
<th style="text-align:right;">
mz
</th>
<th style="text-align:right;">
rt
</th>
<th style="text-align:left;">
compoundDB
</th>
<th style="text-align:left;">
searchTableName
</th>
<th style="text-align:left;">
label
</th>
<th style="text-align:left;">
peak_label
</th>
<th style="text-align:left;">
method_tag
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
Ribose-P
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
Ribose-P
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
OMP
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
OMP
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1,3-diphopshateglycerate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
1,3-diphopshateglycerate
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
3-hydroxy-3-methylglutaryl-CoA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
3-hydroxy-3-methylglutaryl-CoA
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
3-phosphoglycerate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
3-phosphoglycerate
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
6-phospho-D-gluconate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
6-phospho-D-gluconate
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
acetyl-CoA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
acetyl-CoA
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
adenosine
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
adenosine
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<table class="table table table-striped">
<caption>
Samples table
</caption>
<thead><tr>
<th style="text-align:left;">
sampleId
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:left;">
filename
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
NH4_0.055_filter_b
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
NH4_0.055_filter_d
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
NH4_0.055_pellet_a
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
NH4_0.055_pellet_c
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
NH4_0.173_filter_b
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
NH4_0.173_filter_d
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
NH4_0.173_pellet_a
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
NH4_0.173_pellet_c
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
NH4_0.210_filter_b
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
NH4_0.210_filter_d
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
<table class="table table table-striped">
<caption>
Measurements table
</caption>
<thead><tr>
<th style="text-align:left;">
groupId
</th>
<th style="text-align:left;">
sampleId
</th>
<th style="text-align:right;">
log2_abundance
</th>
<th style="text-align:right;">
centered_log2_abundance
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-1.2684248
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-0.8160200
</td>
</tr>
<tr>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-4.5988785
</td>
</tr>
<tr>
<td style="text-align:left;">
25
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-3.2270520
</td>
</tr>
<tr>
<td style="text-align:left;">
26
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-2.2107806
</td>
</tr>
<tr>
<td style="text-align:left;">
30
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-1.1796164
</td>
</tr>
<tr>
<td style="text-align:left;">
40
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-0.3181364
</td>
</tr>
<tr>
<td style="text-align:left;">
51
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-1.0992572
</td>
</tr>
<tr>
<td style="text-align:left;">
56
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-0.4479950
</td>
</tr>
<tr>
<td style="text-align:left;">
57
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:right;">
8.228819
</td>
<td style="text-align:right;">
-3.8019754
</td>
</tr>
</tbody>
</table>
</div>
<div id="add-sample-meta-data" class="section level2">
<h2 class="hasAnchor">
<a href="#add-sample-meta-data" class="anchor"></a>Add sample meta-data</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mzroll_list_augmented</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_samples_tbl.html">merge_samples_tbl</a></span><span class="op">(</span><span class="va">mzroll_list</span>, <span class="va">nplug_samples</span>, <span class="st">"sample_name"</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<caption>
updated mzroll samples table
</caption>
<thead><tr>
<th style="text-align:left;">
sampleId
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:left;">
filename
</th>
<th style="text-align:right;">
samples_tbl_row
</th>
<th style="text-align:left;">
sample_name
</th>
<th style="text-align:left;">
month
</th>
<th style="text-align:left;">
replicate
</th>
<th style="text-align:right;">
DR
</th>
<th style="text-align:left;">
limitation
</th>
<th style="text-align:left;">
exp_ref
</th>
<th style="text-align:left;">
extraction
</th>
<th style="text-align:right;">
condition
</th>
<th style="text-align:right;">
reference
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
NH4_0.055_filter_b
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NH4_0.055_filter_b
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
NH4_0.055_filter_d
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
NH4_0.055_filter_d
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:left;">
D
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
NH4_0.055_pellet_a
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
NH4_0.055_pellet_a
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
NH4_0.055_pellet_c
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
NH4_0.055_pellet_c
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
NH4_0.173_filter_b
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
NH4_0.173_filter_b
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:left;">
B
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
53
</td>
</tr>
</tbody>
</table>
</div>
<div id="add-compounds-meta-data" class="section level2">
<h2 class="hasAnchor">
<a href="#add-compounds-meta-data" class="anchor"></a>Add compounds meta-data</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mzroll_list_augmented</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/merge_compounds_tbl.html">merge_compounds_tbl</a></span><span class="op">(</span><span class="va">mzroll_list_augmented</span>, <span class="va">nplug_compounds</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<caption>
updated mzroll features table
</caption>
<thead><tr>
<th style="text-align:left;">
groupId
</th>
<th style="text-align:left;">
compoundName
</th>
<th style="text-align:left;">
smiles
</th>
<th style="text-align:left;">
adductName
</th>
<th style="text-align:left;">
tagString
</th>
<th style="text-align:right;">
mz
</th>
<th style="text-align:right;">
rt
</th>
<th style="text-align:left;">
compoundDB
</th>
<th style="text-align:left;">
searchTableName
</th>
<th style="text-align:left;">
label
</th>
<th style="text-align:left;">
peak_label
</th>
<th style="text-align:left;">
method_tag
</th>
<th style="text-align:left;">
pathway
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
Ribose-P
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
Ribose-P
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Pentose Phosphate Pathway
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
OMP
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
OMP
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Pyrimidines
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1,3-diphopshateglycerate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
1,3-diphopshateglycerate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Glycolysis
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
3-hydroxy-3-methylglutaryl-CoA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
3-hydroxy-3-methylglutaryl-CoA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Lipid metabolism
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
3-phosphoglycerate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
3-phosphoglycerate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Glycolysis
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="preliminary-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#preliminary-analysis" class="anchor"></a>Preliminary Analysis</h1>
<p>Having loaded the dataset, we can take a quick look at the data using heatmaps.</p>
<div id="limiting-nutrients-and-dilution-rate" class="section level2">
<h2 class="hasAnchor">
<a href="#limiting-nutrients-and-dilution-rate" class="anchor"></a>Limiting Nutrients and Dilution Rate</h2>
<p>The biological variables in the experimental design are limiting nutrient and dilution rate. To look at their influence in shaping the metabolome, we can separate limiting nutrients using facets and order samples based on dilution rate (this will happen naturally if we order alphabetically by sample name).</p>
<p>To make plots like this we can take advantage of the fact that mzroll_lists build on top of the <a href="http://www.github.com/calico/romic"><strong>romic</strong></a> <em>triple_omic</em> classes.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>
  <span class="va">mzroll_list_augmented</span>,
  feature_var <span class="op">=</span> <span class="st">"compoundName"</span>,
  sample_var <span class="op">=</span> <span class="st">"name"</span>,
  value_var <span class="op">=</span> <span class="st">"centered_log2_abundance"</span>,
  change_threshold <span class="op">=</span> <span class="fl">5</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">exp_ref</span> <span class="op">+</span> <span class="va">limitation</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Metabolites separated by limiting nutrients"</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/bio_heatmap-1.png" width="800"></p>
<p>From this plot, its clear that limiting nutrients have a strong role in shaping the metabolome, while dilution rate creates relatively smooth transitions as each limitation progresses from slow growth (left) to fast growth (right).</p>
</div>
<div id="extraction-method-and-batch" class="section level2">
<h2 class="hasAnchor">
<a href="#extraction-method-and-batch" class="anchor"></a>Extraction Method and Batch</h2>
<p>The technical variables in the experimental design are extraction method, month (which is likely associated with batch effects), and replication. Like the heatmap of limitation above, we can look at how extraction method and month shape the metabolome with heatmaps.</p>
<p>Since limiting nutrient has a major effect, we’ll want to account for it when looking extraction method’s impact on metabolome variability.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>
  <span class="va">mzroll_list_augmented</span>,
  feature_var <span class="op">=</span> <span class="st">"compoundName"</span>,
  sample_var <span class="op">=</span> <span class="st">"name"</span>,
  value_var <span class="op">=</span> <span class="st">"centered_log2_abundance"</span>,
  change_threshold <span class="op">=</span> <span class="fl">5</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">limitation</span> <span class="op">+</span> <span class="va">extraction</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Metabolites separated by extraction"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/extraction_heatmap-1.png" width="800"></p>
<p>From this plot, there are clear, but often relatively minor effects of extraction method on metabolite abundances. Large differences in extraction efficiency are generally seen for nucleotides, where using a filter-based extraction results in greater signal than using pellets.</p>
<p>To look at month-effects, we can take advantage of the fact that in each month two biological replicate reference cultures were generated. These two cultures can be distinguished in the sample meta-data most easily by subtle differences in their dilution rate.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mzroll_list_augmented</span> <span class="op">%&gt;%</span>
  <span class="co"># only look at reference samples</span>
  <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/filter_tomic.html">filter_tomic</a></span><span class="op">(</span>
    filter_type <span class="op">=</span> <span class="st">"category"</span>,
    filter_table <span class="op">=</span> <span class="st">"samples"</span>,
    filter_variable <span class="op">=</span> <span class="st">"exp_ref"</span>,
    filter_value <span class="op">=</span> <span class="st">"ref"</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>
  feature_var <span class="op">=</span> <span class="st">"compoundName"</span>,
  sample_var <span class="op">=</span> <span class="st">"name"</span>,
  value_var <span class="op">=</span> <span class="st">"centered_log2_abundance"</span>,
  change_threshold <span class="op">=</span> <span class="fl">5</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">month</span> <span class="op">+</span> <span class="va">DR</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"References separated by month and distinct cultures (by DR)"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/month_heatmap-1.png" width="800"></p>
<p>From this plot, the biological replicate reference cultures are nearly identical (that’s one reason why chemostats are awesome!) while there are subtle differences between months.</p>
</div>
</div>
<div id="normalization" class="section level1">
<h1 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h1>
<p>From the preliminary analysis, when normalizing this dataset to remove sources of technical variability (and thereby clarify the biological signal), we’ll want to account of month-to-month variability, extraction methodology, and technical replication.</p>
<p>To tackle these obstacles we will:</p>
<ol style="list-style-type: decimal">
<li>compare and then collapse technical replicates.</li>
<li>contrast experimental conditions with reference conditions to account for month-to-month variability.</li>
<li>determine whether one extraction method is superior or whether both can be used for downstream analysis with appropriate normalization.</li>
</ol>
<div id="collapsing-technical-replicates" class="section level2">
<h2 class="hasAnchor">
<a href="#collapsing-technical-replicates" class="anchor"></a>Collapsing Technical Replicates</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_compare_injection.html">plot_compare_injection</a></span><span class="op">(</span>
  <span class="va">mzroll_list_augmented</span>,
  grouping_vars <span class="op">=</span> <span class="st">"condition"</span>,
  peak_quant_var <span class="op">=</span> <span class="st">"centered_log2_abundance"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Comparison of injection technical replicates"</span><span class="op">)</span>
<span class="co">#&gt; 3 sample variables will be dropped since they</span>
<span class="co">#&gt; - vary for the same grouping_vars:</span>
<span class="co">#&gt; - samples_tbl_row, sample_name, replicate</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/technical_replicates-1.png" width="800"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mzroll_list_distinct_conditions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/collapse_injections.html">collapse_injections</a></span><span class="op">(</span>
  <span class="va">mzroll_list_augmented</span>,
  grouping_vars <span class="op">=</span> <span class="st">"condition"</span>,
  peak_quant_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"log2_abundance"</span>, <span class="st">"centered_log2_abundance"</span><span class="op">)</span>,
  collapse_fxn <span class="op">=</span> <span class="st">"mean"</span>
  <span class="op">)</span>
<span class="co">#&gt; 3 sample variables will be dropped since they</span>
<span class="co">#&gt; - vary for the same grouping_vars:</span>
<span class="co">#&gt; - samples_tbl_row, sample_name, replicate</span>
<span class="co">#&gt; Warning: `summarise_each_()` was deprecated in dplyr 0.7.0.</span>
<span class="co">#&gt; Please use `across()` instead.</span>
<span class="co">#&gt; Warning: `funs_()` was deprecated in dplyr 0.7.0.</span>
<span class="co">#&gt; Please use `funs()` instead.</span>
<span class="co">#&gt; See vignette('programming') for more help</span>
<span class="co">#&gt; Warning: `funs()` was deprecated in dplyr 0.8.0.</span>
<span class="co">#&gt; Please use a list of either functions or lambdas: </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   # Simple named list: </span>
<span class="co">#&gt;   list(mean = mean, median = median)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   # Auto named with `tibble::lst()`: </span>
<span class="co">#&gt;   tibble::lst(mean, median)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   # Using lambdas</span>
<span class="co">#&gt;   list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))</span></code></pre></div>
<table class="table table table-striped">
<caption>
samples aggregated over injections
</caption>
<thead><tr>
<th style="text-align:right;">
condition
</th>
<th style="text-align:left;">
filename
</th>
<th style="text-align:left;">
month
</th>
<th style="text-align:right;">
DR
</th>
<th style="text-align:left;">
limitation
</th>
<th style="text-align:left;">
exp_ref
</th>
<th style="text-align:left;">
extraction
</th>
<th style="text-align:right;">
reference
</th>
<th style="text-align:left;">
sampleId
</th>
<th style="text-align:left;">
name
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
NH4_0.055_filter_b
</td>
</tr>
<tr>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
NH4_0.055_pellet_a
</td>
</tr>
<tr>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
NH4_0.173_filter_b
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
NH4_0.173_pellet_a
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.210
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
NH4_0.210_filter_b
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.210
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
NH4_0.210_pellet_a
</td>
</tr>
<tr>
<td style="text-align:right;">
7
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.290
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
NH4_0.290_filter_b
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.290
</td>
<td style="text-align:left;">
NH4
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
NH4_0.290_pellet_a
</td>
</tr>
<tr>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:left;">
URA
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
filter
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
URA_0.056_filter_b
</td>
</tr>
<tr>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Jun
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:left;">
URA
</td>
<td style="text-align:left;">
exp
</td>
<td style="text-align:left;">
pellet
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
URA_0.056_pellet_a
</td>
</tr>
</tbody>
</table>
</div>
<div id="contrasting-to-a-reference-condition" class="section level2">
<h2 class="hasAnchor">
<a href="#contrasting-to-a-reference-condition" class="anchor"></a>Contrasting to a Reference Condition</h2>
<p>Since experimental cultures of the same limiting nutrient were generated simultaneously, limitation and the month-of-generation are confounded. Without correcting this confounding, some of the month effects would be interpreted as limitation-specific signals. To correct for this month-to-month variability in instrument performance, each experimental sample can be compared to common reference conditions. Since these cultures possess very similar biological signals, but should share the performance biases, contrasting each biological culture with a month-matched reference should remove this shared bias. In a similar fashion we can evaluate whether differences due to extraction methodology are resolved when we are comparing an experimental condition to its extraction-matched, month-matched reference.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mzroll_list_normalized</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_peaks.html">normalize_peaks</a></span><span class="op">(</span>
  <span class="va">mzroll_list_distinct_conditions</span>,
  normalization_method <span class="op">=</span> <span class="st">"reference sample"</span>,
  quant_peak_varname <span class="op">=</span> <span class="st">"log2_abundance"</span>,
  norm_peak_varname <span class="op">=</span> <span class="st">"normalized_log2_abundance"</span>,
  batch_varnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"month"</span>, <span class="st">"extraction"</span><span class="op">)</span>,
  reference_varname <span class="op">=</span> <span class="st">"exp_ref"</span>,
  reference_values <span class="op">=</span> <span class="st">"ref"</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># having normalized by the common reference, we can re-center the data</span>
  <span class="co">#  since slow-phosphate limited growth is not a biological reference.</span>
  <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/center_tomic.html">center_tomic</a></span><span class="op">(</span>measurement_vars <span class="op">=</span> <span class="st">"normalized_log2_abundance"</span><span class="op">)</span></code></pre></div>
<p>Have normalized for month and extraction method, we can look at the results as another heatmap.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>
  <span class="va">mzroll_list_normalized</span>,
  feature_var <span class="op">=</span> <span class="st">"compoundName"</span>,
  sample_var <span class="op">=</span> <span class="st">"name"</span>,
  value_var <span class="op">=</span> <span class="st">"normalized_log2_abundance"</span>,
  change_threshold <span class="op">=</span> <span class="fl">5</span>,
  cluster_dim <span class="op">=</span> <span class="st">"rows"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">exp_ref</span> <span class="op">+</span> <span class="va">limitation</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Post-normalization metabolite abundances"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/post_normalized_heatmap-1.png" width="800"></p>
<p>From this visualization, we can note that the reference chemostats are uniform for a given peakgroup (these were set to zero during normalization and then offset during re-centering). There does seem to be an issue with the attempted extraction normalization though; having compared each experimental sample to its extraction method matched reference, meaningful differences relative abundance associated with extraction remain.</p>
</div>
<div id="accounting-for-extraction-method" class="section level2">
<h2 class="hasAnchor">
<a href="#accounting-for-extraction-method" class="anchor"></a>Accounting for Extraction Method</h2>
<p>While we could try to make use of data from both extraction methods, as they did in the original paper, due to the inconsistent extraction efficiency, I think its better to just retain the higher-signal filter-extraction samples. At this point we can also discard the reference samples since they’re no longer needed.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">final_processed_data</span> <span class="op">&lt;-</span> <span class="va">mzroll_list_normalized</span> <span class="op">%&gt;%</span>
  <span class="co"># retain just experimental samples</span>
  <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/filter_tomic.html">filter_tomic</a></span><span class="op">(</span>
    filter_type <span class="op">=</span> <span class="st">"category"</span>,
    filter_table <span class="op">=</span> <span class="st">"samples"</span>,
    filter_variable <span class="op">=</span> <span class="st">"exp_ref"</span>,
    filter_value <span class="op">=</span> <span class="st">"exp"</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># retain only filter extraction</span>
  <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/filter_tomic.html">filter_tomic</a></span><span class="op">(</span>
    filter_type <span class="op">=</span> <span class="st">"category"</span>,
    filter_table <span class="op">=</span> <span class="st">"samples"</span>,
    filter_variable <span class="op">=</span> <span class="st">"extraction"</span>,
    filter_value <span class="op">=</span> <span class="st">"filter"</span>
    <span class="op">)</span>
  
<span class="co"># clean-up sample data</span>
<span class="va">renamed_samples</span> <span class="op">&lt;-</span> <span class="va">final_processed_data</span><span class="op">$</span><span class="va">samples</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">sampleId</span>, <span class="va">limitation</span>, <span class="va">DR</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>name <span class="op">=</span> <span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html">glue</a></span><span class="op">(</span>
    <span class="st">"{stringr::str_sub(limitation, 1, 1)}{round(DR,2)}"</span>
  <span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
    name <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span> <span class="op">~</span> <span class="va">name</span>,
                     <span class="cn">TRUE</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">name</span>, <span class="st">"-"</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span>
   
<span class="va">final_processed_data</span> <span class="op">&lt;-</span> <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/update_tomic.html">update_tomic</a></span><span class="op">(</span>
  <span class="va">final_processed_data</span>,
  <span class="va">renamed_samples</span>
  <span class="op">)</span>

<span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>
  <span class="va">final_processed_data</span>,
  feature_var <span class="op">=</span> <span class="st">"compoundName"</span>,
  sample_var <span class="op">=</span> <span class="st">"name"</span>,
  value_var <span class="op">=</span> <span class="st">"normalized_log2_abundance"</span>,
  change_threshold <span class="op">=</span> <span class="fl">5</span>,
  cluster_dim <span class="op">=</span> <span class="st">"rows"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span><span class="op">~</span> <span class="va">limitation</span>, scales <span class="op">=</span> <span class="st">"free_x"</span>, space <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Final processed metabolite abundances across 25 growth conditions"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in if (class(distinct_samples[[sample_var]]) %in% c("factor",</span>
<span class="co">#&gt; "ordered")) {: the condition has length &gt; 1 and only the first element will be</span>
<span class="co">#&gt; used</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/extraction_correction-1.png" width="800"></p>
</div>
</div>
<div id="differential-abundance-testing" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-abundance-testing" class="anchor"></a>Differential Abundance Testing</h1>
<p>Having generated a processed dataset, we can already see some patterns clearly by looking at the heatmap, such as bottlenecking of pyrimidine biosynthesis during uracil limitation, amino acid accumulation during leucine limitation, and TCA up-regulation during ammonium limitation. Its also apparent that many of the strongest metabolomic alterations are observed at slow growth rates, while these there are relatively smooth transitions back to a normalish-metabolome as cell growth more quickly. To quantitatively summarize trends like these, its helpful to carryout differential abundance testing. But, before we proceed we should determine the type of model we need to fit by interpreting the major sources of variation in our dataset in light of the experimental design.</p>
<div id="exploratory-data-analysis-eda" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-data-analysis-eda" class="anchor"></a>Exploratory Data Analysis (EDA)</h2>
<p>Our analysis thus far has already corrected for major sources of variation in this original dataset, such as month-to-month variation and differences in extraction methodology. If we hadn’t controlled for those factors by this point, we would want to create visualizations which appropriately identify major unaccounted for sources of variation. Similarly, this will help us to determine how we should represent our experimental design as an appropriate hypothesis test.</p>
<p>Two visualizations that are particularly useful for EDA are heatmaps (I’m sure you’ve seen enough of those), and principal components plots which overlay elements of the experimental design on the leading principal components.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samples_with_pcs</span> <span class="op">&lt;-</span> <span class="va">final_processed_data</span> <span class="op">%&gt;%</span>
  <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/add_pca_loadings.html">add_pca_loadings</a></span><span class="op">(</span>value_var <span class="op">=</span> <span class="st">"normalized_log2_abundance"</span>, npcs <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>

<span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/plot_bivariate.html">plot_bivariate</a></span><span class="op">(</span>
  <span class="va">samples_with_pcs</span><span class="op">$</span><span class="va">samples</span>,
  <span class="st">"PC1"</span>,
  <span class="st">"PC2"</span>,
  color_var <span class="op">=</span> <span class="st">"limitation"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Top principal components driving metabolomic variation"</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/static_eda-1.png" width="800"></p>
<p>One plot doesn’t quite cut it here. Luckily, romic has some powerful methods for shiny-based interactive analysis that we can leverage here to quickly make plots like these.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/app_flow.html">app_flow</a></span><span class="op">(</span><span class="va">samples_with_pcs</span><span class="op">)</span>

<span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/app_heatmap.html">app_heatmap</a></span><span class="op">(</span><span class="va">final_processed_data</span> <span class="op">%&gt;%</span> <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/triple_to_tidy.html">triple_to_tidy</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="hypothesis-testing" class="section level2">
<h2 class="hasAnchor">
<a href="#hypothesis-testing" class="anchor"></a>Hypothesis Testing</h2>
<p>From the results thus far, we’ve seen that limitation greatly impacts the metabolome, as does growth rate which tends to have a limitation-specific trend. Whatever model we fit should account for these features of the data. The actual model we fit will depend on the question we want to ask of the data. For example, we could be interested in just finding metabolites impacted by nutrient, or those that are perturbed by a specific nutrient. To keep things manageable, I’ll explore how nitrogen-limitation alters the metatbolome.</p>
<p>To ask this, we can formulate a linear regression of the form:</p>
<p><span class="math display">\[
y^{\text{normalized}} \sim \text{lim} + \text{lim} \cdot \text{DR}
\]</span></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">regression_significance</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diffex_mzroll.html">diffex_mzroll</a></span><span class="op">(</span>
  <span class="va">final_processed_data</span>,
  <span class="st">"normalized_log2_abundance"</span>,
  <span class="st">"limitation + limitation:DR + 0"</span>
  <span class="op">)</span>
<span class="co">#&gt; Warning in .f(.x[[i]], ...): q-value calculation initially failed due to too many small p-values</span>
<span class="co">#&gt;         but claman was able to recover results</span>

<span class="fu"><a href="../reference/plot_pvalues.html">plot_pvalues</a></span><span class="op">(</span><span class="va">regression_significance</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/hypothesis_testing-1.png" width="800"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">regression_significance</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/hypothesis_testing-2.png" width="800"></p>
</div>
<div id="plotting-individual-metabolites" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-individual-metabolites" class="anchor"></a>Plotting Individual Metabolites</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">n_lim_signif</span> <span class="op">&lt;-</span> <span class="va">regression_significance</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">term</span> <span class="op">==</span> <span class="st">"limitationNH4"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">qvalue</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span>
    <span class="va">final_processed_data</span><span class="op">$</span><span class="va">features</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">groupId</span>, <span class="va">peak_label</span><span class="op">)</span>,
    by <span class="op">=</span> <span class="st">"groupId"</span><span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<caption>
Metabolites that most perturbed by nitrogen limitation
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
groupId
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
diffex_label
</th>
<th style="text-align:left;">
peak_label
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:right;">
-4.653987
</td>
<td style="text-align:right;">
0.3092707
</td>
<td style="text-align:right;">
-15.048264
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
-15.048 ***
</td>
<td style="text-align:left;">
glutamine
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
81
</td>
<td style="text-align:right;">
-3.148848
</td>
<td style="text-align:right;">
0.2523321
</td>
<td style="text-align:right;">
-12.478986
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
-12.479 ***
</td>
<td style="text-align:left;">
serine
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:right;">
-1.437548
</td>
<td style="text-align:right;">
0.1200230
</td>
<td style="text-align:right;">
-11.977270
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
-11.977 ***
</td>
<td style="text-align:left;">
OMP
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
74
</td>
<td style="text-align:right;">
2.950303
</td>
<td style="text-align:right;">
0.2765627
</td>
<td style="text-align:right;">
10.667753
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:left;">
10.668 ***
</td>
<td style="text-align:left;">
phenylpyruvate
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
86
</td>
<td style="text-align:right;">
3.706008
</td>
<td style="text-align:right;">
0.3477182
</td>
<td style="text-align:right;">
10.658078
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:left;">
10.658 ***
</td>
<td style="text-align:left;">
trehalose/sucrose
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:right;">
-4.904499
</td>
<td style="text-align:right;">
0.5012941
</td>
<td style="text-align:right;">
-9.783677
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
3.0e-07
</td>
<td style="text-align:left;">
-9.784 ***
</td>
<td style="text-align:left;">
arginine
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
59
</td>
<td style="text-align:right;">
-2.738652
</td>
<td style="text-align:right;">
0.2782964
</td>
<td style="text-align:right;">
-9.840773
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
3.0e-07
</td>
<td style="text-align:left;">
-9.841 ***
</td>
<td style="text-align:left;">
leucine/isoleucine
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
54
</td>
<td style="text-align:right;">
-4.324299
</td>
<td style="text-align:right;">
0.5146966
</td>
<td style="text-align:right;">
-8.401647
</td>
<td style="text-align:right;">
3e-07
</td>
<td style="text-align:right;">
1.9e-06
</td>
<td style="text-align:left;">
-8.402 ***
</td>
<td style="text-align:left;">
histidine
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
88
</td>
<td style="text-align:right;">
-2.803336
</td>
<td style="text-align:right;">
0.3345180
</td>
<td style="text-align:right;">
-8.380224
</td>
<td style="text-align:right;">
3e-07
</td>
<td style="text-align:right;">
1.9e-06
</td>
<td style="text-align:left;">
-8.38 ***
</td>
<td style="text-align:left;">
tryptophan
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:right;">
-7.261097
</td>
<td style="text-align:right;">
0.9072313
</td>
<td style="text-align:right;">
-8.003578
</td>
<td style="text-align:right;">
6e-07
</td>
<td style="text-align:right;">
3.0e-06
</td>
<td style="text-align:left;">
-8.004 ***
</td>
<td style="text-align:left;">
citrulline
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_barplot.html">plot_barplot</a></span><span class="op">(</span>
  mzroll_list <span class="op">=</span> <span class="va">final_processed_data</span>,
  groupIds <span class="op">=</span> <span class="va">n_lim_signif</span><span class="op">$</span><span class="va">groupId</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,
  grouping_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"limitation"</span>, <span class="st">"DR"</span><span class="op">)</span>,
  value_var <span class="op">=</span> <span class="st">"normalized_log2_abundance"</span>,
  fill_var <span class="op">=</span> <span class="st">"limitation"</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set2"</span><span class="op">)</span></code></pre></div>
<p><img src="NPLUG_files/figure-html/met_examples_barplot-1.png" width="800"></p>
</div>
<div id="pathway-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#pathway-analysis" class="anchor"></a>Pathway Analysis</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathway_nest</span> <span class="op">&lt;-</span> <span class="va">final_processed_data</span><span class="op">$</span><span class="va">features</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">groupId</span>, <span class="va">pathway</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html">nest</a></span><span class="op">(</span>pathway_members <span class="op">=</span> <span class="va">groupId</span><span class="op">)</span>

<span class="va">pathway_list</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span>
  <span class="va">pathway_nest</span><span class="op">$</span><span class="va">pathway_members</span>,
    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">groupId</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pathway_list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">pathway_nest</span><span class="op">$</span><span class="va">pathway</span>

<span class="va">enrichments</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_pathway_enrichments.html">find_pathway_enrichments</a></span><span class="op">(</span>
  <span class="va">final_processed_data</span>,
  <span class="va">regression_significance</span>,
  <span class="va">pathway_list</span>,
  test_absolute_effects <span class="op">=</span> <span class="cn">FALSE</span>
  <span class="op">)</span></code></pre></div>
<div id="generate-tables-and-plots-of-top-enrichments" class="section level3">
<h3 class="hasAnchor">
<a href="#generate-tables-and-plots-of-top-enrichments" class="anchor"></a>Generate tables and plots of top enrichments</h3>
<table class="table table table-striped">
<caption>
Top pathway enrichments across all regression terms
</caption>
<thead><tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:left;">
pathway
</th>
<th style="text-align:right;">
pval
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:right;">
ES
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:right;">
size
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
limitationNH4:DR
</td>
<td style="text-align:left;">
Amino Acids
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
0.0000006
</td>
<td style="text-align:right;">
0.8716301
</td>
<td style="text-align:right;">
2.108136
</td>
<td style="text-align:right;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
Amino Acids
</td>
<td style="text-align:right;">
0.0000023
</td>
<td style="text-align:right;">
0.0000226
</td>
<td style="text-align:right;">
-0.8381768
</td>
<td style="text-align:right;">
-2.043089
</td>
<td style="text-align:right;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
TCA cycle
</td>
<td style="text-align:right;">
0.0007502
</td>
<td style="text-align:right;">
0.0037511
</td>
<td style="text-align:right;">
0.8410193
</td>
<td style="text-align:right;">
2.017484
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationLEU
</td>
<td style="text-align:left;">
Amino Acids
</td>
<td style="text-align:right;">
0.0007378
</td>
<td style="text-align:right;">
0.0073784
</td>
<td style="text-align:right;">
0.7185995
</td>
<td style="text-align:right;">
1.773362
</td>
<td style="text-align:right;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4:DR
</td>
<td style="text-align:left;">
TCA cycle
</td>
<td style="text-align:right;">
0.0025193
</td>
<td style="text-align:right;">
0.0125966
</td>
<td style="text-align:right;">
-0.7638691
</td>
<td style="text-align:right;">
-1.987874
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4
</td>
<td style="text-align:left;">
Glycolysis
</td>
<td style="text-align:right;">
0.0091217
</td>
<td style="text-align:right;">
0.0304055
</td>
<td style="text-align:right;">
0.7275549
</td>
<td style="text-align:right;">
1.886448
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationPO4
</td>
<td style="text-align:left;">
Purine Synthesis
</td>
<td style="text-align:right;">
0.0040686
</td>
<td style="text-align:right;">
0.0406858
</td>
<td style="text-align:right;">
0.7323962
</td>
<td style="text-align:right;">
1.738962
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationNH4:DR
</td>
<td style="text-align:left;">
Purine Synthesis
</td>
<td style="text-align:right;">
0.0193462
</td>
<td style="text-align:right;">
0.0644873
</td>
<td style="text-align:right;">
-0.5833708
</td>
<td style="text-align:right;">
-1.699125
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationPO4:DR
</td>
<td style="text-align:left;">
Glycolysis
</td>
<td style="text-align:right;">
0.0159305
</td>
<td style="text-align:right;">
0.0796526
</td>
<td style="text-align:right;">
0.7327704
</td>
<td style="text-align:right;">
1.652984
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
limitationPO4:DR
</td>
<td style="text-align:left;">
Purine Synthesis
</td>
<td style="text-align:right;">
0.0128088
</td>
<td style="text-align:right;">
0.0796526
</td>
<td style="text-align:right;">
-0.7096962
</td>
<td style="text-align:right;">
-1.692692
</td>
<td style="text-align:right;">
12
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># see specific pathway enrichments</span>
<span class="va">ranked_nitrogen_enrichments</span> <span class="op">&lt;-</span> <span class="va">enrichments</span><span class="op">$</span><span class="va">enrichment_table</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">padj</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">term</span> <span class="op">==</span> <span class="st">"limitationNH4"</span><span class="op">)</span>

<span class="va">ranked_nitrogen_enrichments</span><span class="op">$</span><span class="va">enrichment_plot</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span>
    <span class="st">"Amino acids are depleted during nitrogen limitation"</span>,
    <span class="st">"Note that the x-axis is ranks, and negative values have higher ranks"</span>
    <span class="op">)</span> </code></pre></div>
<p><img src="NPLUG_files/figure-html/top_enrichments-1.png" width="800"></p>
</div>
</div>
</div>
<div id="export-results" class="section level1">
<h1 class="hasAnchor">
<a href="#export-results" class="anchor"></a>Export results</h1>
<p>While self-contained reproducible notebooks like this vignette are an ideal way to share results with collaborators, normalized data and analysis artifacts can easily be exported as well. This is done using romic functions: <em>export_tomic_as_tidy</em>, <em>export_tomic_as_triple</em>, and <em>export_tomic_as_wide</em>. These three exports write the same comprehensive summaries of an mzroll to disk just in different formats. Before exporting, we may also want to include additional feature and sample attributes that have not been directly incorporated into the mzroll_list such as summaries of differential expression.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">wide_stats</span> <span class="op">&lt;-</span> <span class="va">regression_significance</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">groupId</span>, <span class="va">term</span>, <span class="va">diffex_label</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/spread.html">spread</a></span><span class="op">(</span><span class="va">term</span>, <span class="va">diffex_label</span><span class="op">)</span>

<span class="va">final_processed_data</span> <span class="op">&lt;-</span> <span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/update_tomic.html">update_tomic</a></span><span class="op">(</span>
  <span class="va">final_processed_data</span>,
  <span class="va">final_processed_data</span><span class="op">$</span><span class="va">features</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">wide_stats</span>, by <span class="op">=</span> <span class="st">"groupId"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<table class="table table table-striped">
<caption>
Peakgroups with differential abundance statistics added
</caption>
<thead><tr>
<th style="text-align:left;">
groupId
</th>
<th style="text-align:left;">
compoundName
</th>
<th style="text-align:left;">
smiles
</th>
<th style="text-align:left;">
adductName
</th>
<th style="text-align:left;">
tagString
</th>
<th style="text-align:right;">
mz
</th>
<th style="text-align:right;">
rt
</th>
<th style="text-align:left;">
compoundDB
</th>
<th style="text-align:left;">
searchTableName
</th>
<th style="text-align:left;">
label
</th>
<th style="text-align:left;">
peak_label
</th>
<th style="text-align:left;">
method_tag
</th>
<th style="text-align:left;">
pathway
</th>
<th style="text-align:left;">
limitationGLU
</th>
<th style="text-align:left;">
limitationGLU:DR
</th>
<th style="text-align:left;">
limitationLEU
</th>
<th style="text-align:left;">
limitationLEU:DR
</th>
<th style="text-align:left;">
limitationNH4
</th>
<th style="text-align:left;">
limitationNH4:DR
</th>
<th style="text-align:left;">
limitationPO4
</th>
<th style="text-align:left;">
limitationPO4:DR
</th>
<th style="text-align:left;">
limitationURA
</th>
<th style="text-align:left;">
limitationURA:DR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
Ribose-P
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
Ribose-P
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Pentose Phosphate Pathway
</td>
<td style="text-align:left;">
1.396
</td>
<td style="text-align:left;">
3.737 **
</td>
<td style="text-align:left;">
1.369
</td>
<td style="text-align:left;">
2.986 *
</td>
<td style="text-align:left;">
1.052
</td>
<td style="text-align:left;">
3.058 *
</td>
<td style="text-align:left;">
-15.989 ***
</td>
<td style="text-align:left;">
15.46 ***
</td>
<td style="text-align:left;">
-2.954 **
</td>
<td style="text-align:left;">
5.917 ***
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
OMP
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
OMP
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Pyrimidines
</td>
<td style="text-align:left;">
-12.467 ***
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
-11.782 ***
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
-11.977 ***
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
-14.86 ***
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
59.939 ***
</td>
<td style="text-align:left;">
6.303 ***
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1,3-diphopshateglycerate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
1,3-diphopshateglycerate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Glycolysis
</td>
<td style="text-align:left;">
-0.016
</td>
<td style="text-align:left;">
1.49
</td>
<td style="text-align:left;">
-1.785 *
</td>
<td style="text-align:left;">
2.223 *
</td>
<td style="text-align:left;">
-1.722 *
</td>
<td style="text-align:left;">
3.465 **
</td>
<td style="text-align:left;">
-5.349 ***
</td>
<td style="text-align:left;">
5.63 ***
</td>
<td style="text-align:left;">
-1.417 *
</td>
<td style="text-align:left;">
2.486 *
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
3-hydroxy-3-methylglutaryl-CoA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
3-hydroxy-3-methylglutaryl-CoA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Lipid metabolism
</td>
<td style="text-align:left;">
1.196
</td>
<td style="text-align:left;">
-0.454
</td>
<td style="text-align:left;">
2.859 *
</td>
<td style="text-align:left;">
-0.623
</td>
<td style="text-align:left;">
-3.007 **
</td>
<td style="text-align:left;">
0.425
</td>
<td style="text-align:left;">
0.123
</td>
<td style="text-align:left;">
-0.382
</td>
<td style="text-align:left;">
3.087 **
</td>
<td style="text-align:left;">
-3.193 *
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
3-phosphoglycerate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
3-phosphoglycerate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Glycolysis
</td>
<td style="text-align:left;">
8.485 ***
</td>
<td style="text-align:left;">
-2.293 *
</td>
<td style="text-align:left;">
-1.218
</td>
<td style="text-align:left;">
1.438
</td>
<td style="text-align:left;">
7.28 ***
</td>
<td style="text-align:left;">
0.044
</td>
<td style="text-align:left;">
-9.049 ***
</td>
<td style="text-align:left;">
5.192 ***
</td>
<td style="text-align:left;">
-4.438 ***
</td>
<td style="text-align:left;">
2.813 *
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
6-phospho-D-gluconate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
6-phospho-D-gluconate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Pentose Phosphate Pathway
</td>
<td style="text-align:left;">
3.228 **
</td>
<td style="text-align:left;">
0.037
</td>
<td style="text-align:left;">
1.28
</td>
<td style="text-align:left;">
0.457
</td>
<td style="text-align:left;">
-0.012
</td>
<td style="text-align:left;">
1.148
</td>
<td style="text-align:left;">
-6.988 ***
</td>
<td style="text-align:left;">
7.014 ***
</td>
<td style="text-align:left;">
-0.108
</td>
<td style="text-align:left;">
2.115 *
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
acetyl-CoA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
acetyl-CoA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
TCA cycle
</td>
<td style="text-align:left;">
-1.912 *
</td>
<td style="text-align:left;">
2.686 *
</td>
<td style="text-align:left;">
2.951 *
</td>
<td style="text-align:left;">
-0.105
</td>
<td style="text-align:left;">
-0.565
</td>
<td style="text-align:left;">
0.675
</td>
<td style="text-align:left;">
-1.405 *
</td>
<td style="text-align:left;">
1.791 *
</td>
<td style="text-align:left;">
3.085 **
</td>
<td style="text-align:left;">
-0.579
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
aconitate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
TCA cycle
</td>
<td style="text-align:left;">
-1.687 *
</td>
<td style="text-align:left;">
-2.015
</td>
<td style="text-align:left;">
-0.027
</td>
<td style="text-align:left;">
-2.752 *
</td>
<td style="text-align:left;">
2.702 *
</td>
<td style="text-align:left;">
1.761 *
</td>
<td style="text-align:left;">
3.219 **
</td>
<td style="text-align:left;">
-2.971 *
</td>
<td style="text-align:left;">
-1.366 *
</td>
<td style="text-align:left;">
2.523 *
</td>
</tr>
<tr>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
adenosine
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
adenosine
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Purine Synthesis
</td>
<td style="text-align:left;">
-8.111 ***
</td>
<td style="text-align:left;">
0.558
</td>
<td style="text-align:left;">
1.565
</td>
<td style="text-align:left;">
-5.245 **
</td>
<td style="text-align:left;">
-4.461 ***
</td>
<td style="text-align:left;">
-1.354
</td>
<td style="text-align:left;">
12.063 ***
</td>
<td style="text-align:left;">
-6.701 ***
</td>
<td style="text-align:left;">
0.416
</td>
<td style="text-align:left;">
-2.734 *
</td>
</tr>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
g
</td>
<td style="text-align:left;">
ADP
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
Energetics
</td>
<td style="text-align:left;">
-2.185 *
</td>
<td style="text-align:left;">
0.629
</td>
<td style="text-align:left;">
1.621 *
</td>
<td style="text-align:left;">
-1.196
</td>
<td style="text-align:left;">
0.679
</td>
<td style="text-align:left;">
-0.84
</td>
<td style="text-align:left;">
-2.611 **
</td>
<td style="text-align:left;">
2.362 *
</td>
<td style="text-align:left;">
1.191 *
</td>
<td style="text-align:left;">
-0.903
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">romic</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/romic/man/export_tomic_as_tidy.html">export_tomic_as_tidy</a></span><span class="op">(</span>
  <span class="va">final_processed_data</span>,
  dir_path <span class="op">=</span> <span class="st">"/tmp"</span>,
  name_preamble <span class="op">=</span> <span class="st">"nplug"</span>
  <span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sean Hackett, Phil Seitzer, Calico Life Sciences LLC.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
